webhook_to_fedora_messaging_messages/forgejo/v1.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-FileCopyrightText: 2024-2025 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+import json
+import typing
+from hashlib import sha1, sha256
+
+from ..base import Webhook2FedMsgBase
+from .utils import summarize_repository_event
+
+
+class ForgejoMessageV1(Webhook2FedMsgBase):
+    @property
+    def app_name(self) -> str:
+        """Application name on Fedora Messaging"""
+        return "Forgejo"
+
+    @property
+    def target_id(self) -> str:
+        """Webhook identifier specific to Forgejo"""
+        return self.body["body"]["repository"]["id"]
+
+    @property
+    def target_type(self) -> str:
+        """Webhook type specific to Forgejo - Computed"""
+        return "repository"
+
+    @property
+    def delivery(self) -> str:
+        """Unique identifier (GUID) for event"""
+        return self.body["headers"]["x-forgejo-delivery"]
+
+    @property
+    def signature(self) -> str:
+        """SHA160 signature of the request - Computed"""
+        return (
+            "sha1="
+            + sha1(  # noqa: S324
+                json.dumps(self.body, sort_keys=True, separators=(",", ":")).encode(
+                    encoding="utf-8"
+                )
+            ).hexdigest()
+        )
+
+    @property
+    def signature_sha256(self) -> str:
+        """SHA256 signature of the request - Computed"""
+        return (
+            "sha256="
+            + sha256(
+                json.dumps(self.body, sort_keys=True, separators=(",", ":")).encode(
```

Comment:
This is reused from the other signature and could be factored out for future use in `utils.py`, maybe as a `serialize_body()` function?